### PR TITLE
[`flake8-bugbear`] Allow tuples of exceptions (`B030`)

### LINKED ISSUE #10426: Bug in B030: False positive when adding tuples of exception classes
Hello :slightly_smiling_face: 

When enabling rule B030, these cases are being flagged as violated:

```python
try:
    ...
except (ValueError,TypeError) + (EOFError,ArithmeticError):
    ...
```

```python
BASE_EXCEPTIONS = (ValueError,TypeError)

try:
    ...
except BASE_EXCEPTIONS + (EOFError,ArithmeticError):
    ...
```

Ruff version 0.3.2

### PR DESCRIPTION
**Pull request by ottaviohartman**
Fixes #10426 

## Summary

Fix rule B030 giving a false positive with Tuple operations like `+`.

[Playground](https://play.ruff.rs/17b086bc-cc43-40a7-b5bf-76d7d5fce78a)
```python
try:
    ...
except (ValueError,TypeError) + (EOFError,ArithmeticError):
    ...
```

## Reviewer notes

This is a little more convoluted than I was expecting -- because we can have valid nested Tuples with operations done on them, the flattening logic has become a bit more complex.

Shall I guard this behind --preview?

## Test Plan

Unit tested.

### CODE REVIEW

**Comment by autinerd** [comment]
Oh, we seem to have worked on it at the same time, I just wanted to open a PR fixing this as well :sweat_smile: 

I checked out your PR and checked it against my tests, and it doesn't seem to be complete with the nesting.

```python
try:
    ...
except (ValueError, *(RuntimeError, TypeError), *((ArithmeticError,) + (EOFError,))):
                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ B030
    ...
```

This is flagged as error, although this is valid.

**Comment by ottaviohartman** [comment]
Ok I can close this in favor of yours.

**Comment by ottaviohartman** [comment]
@autinerd actually let me know if you think I should fix that bug and continue, or just close this! Sorry about the confusion - I meant to say that I'd work on this issue.

**Comment by autinerd** [comment]
I think that would be a good idea if you work on it, as my knowledge in the Ruff codebase (and Rust in general) is not really good. My solution was to handle it recursively, maybe it is possible to do it iteratively as well.

File: crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B030.py [hunk 1]
```diff
@@ -76,3 +83,45 @@ def what_to_catch():
...
+try:
+    pass
+except a + (b, c):  # ok
+    pass
+
+
+try:
+    pass
+except (ValueError, *(RuntimeError, TypeError), *((ArithmeticError,) + (EOFError,))):
+    pass
```
Comment (ottaviohartman) [inline crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B030.py:121]: @autinerd added this test case

**Review by charliermarsh — APPROVED** [review]
This looks good! Thanks.

File: crates/ruff_linter/src/rules/flake8_bugbear/rules/except_with_non_exception_classes.rs [hunk 2]
```diff
@@ -89,3 +65,61 @@ pub(crate) fn except_with_non_exception_classes(
...
+                    op: Operator::Add, ..
+                }) => {
+                    exprs_to_process.push_back(value);
+                }
+                _ => flattened_exprs.push(value),
+            },
+            Expr::BinOp(ast::ExprBinOp {
+                left,
+                right,
+                op: Operator::Add,
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/flake8_bugbear/rules/except_with_non_exception_classes.rs:99]: Gated these to `Operator::Add`.

**Comment by charliermarsh** [comment]
No need for preview here since it's a bug fix.

**Pull request merged**